### PR TITLE
MAINT: replace black, flake8 and isort by ruff

### PR DESCRIPTION
**Pull request by theroggy**
Same approach as used in geopandas and pyogrio.

Didn't enable pydocstyle yet to keep the PR somewhat managable.

### CODE REVIEW

File: shapely/tests/geometry/test_equality.py [hunk 1]
```diff
@@ -232,16 +235,16 @@ def test_comparison_not_supported():
...
-        geom1 < geom2
+        assert geom1 < geom2
 
     with pytest.raises(TypeError, match="not supported between instances"):
-        geom1 >= geom2
+        assert geom1 >= geom2
 
     with pytest.raises(TypeError, match="not supported between instances"):
-        geom1 <= geom2
+        assert geom1 <= geom2
```
Comment (mwtoews) [inline shapely/tests/geometry/test_equality.py:247]: Plenty of good fixes like this in here.

File: shapely/tests/test_set_operations.py [hunk 2]
```diff
@@ -334,8 +335,8 @@ def test_coverage_union_non_polygon_inputs(geom_1, geom_2):
         def effective_geom_types(geom):
             if hasattr(geom, "geoms") and not geom.is_empty:
                 gts = set()
-                for geom in geom.geoms:
-                    gts |= effective_geom_types(geom)
+                for part in geom.geoms:
+                    gts |= effective_geom_types(part)
```
Comment (mwtoews) [inline shapely/tests/test_set_operations.py:339]: More good catches!

**Review by mwtoews — APPROVED** [review]
Thanks for PR! I've recently been switching other projects to ruff too, but you have beat me to it!

File: shapely/tests/geometry/test_equality.py (hunk 1 above)
Comment (jorisvandenbossche) [reply shapely/tests/geometry/test_equality.py:247]: Why is this actually needed? We should never get to that `assert`, as the code is meant to raise an error (I don't think I ever add explicit asserts when using `with pytest.raises` contexts)

File: shapely/tests/test_ragged_array.py [hunk 3]
```diff
@@ -317,8 +318,10 @@ def test_multilinestrings():
 def test_multipolygons():
     arr = shapely.from_wkt(
         [
-            "MULTIPOLYGON (((35 10, 45 45, 15 40, 10 20, 35 10), (20 30, 35 35, 30 20, 20 30)))",
-            "MULTIPOLYGON (((40 40, 20 45, 45 30, 40 40)),((20 35, 10 30, 10 10, 30 5, 45 20, 20 35),(30 20, 20 15, 20 25, 30 20)))",
+            "MULTIPOLYGON (((35 10, 45 45, 15 40, 10 20, 35 10), "
+            "(20 30, 35 35, 30 20, 20 30)))",
+            "MULTIPOLYGON (((40 40, 20 45, 45 30, 40 40)),"
+            "((20 35, 10 30, 10 10, 30 5, 45 20, 20 35),(30 20, 20 15, 20 25, 30 20)))",
             "MULTIPOLYGON EMPTY",
```
Comment (jorisvandenbossche) [inline shapely/tests/test_ragged_array.py:325]: For cases like this a `  # noqa: E501` might be nicer?

**Review by jorisvandenbossche — COMMENTED** [review]
Thanks for doing this!

File: shapely/tests/geometry/test_equality.py (hunk 1 above)
Comment (theroggy) [reply shapely/tests/geometry/test_equality.py:247]: Not really needed in this specific instance, but in most cases this is a bug, so ruff gives a linting error on it...

File: shapely/tests/test_ragged_array.py (hunk 3 above)
Comment (theroggy) [reply shapely/tests/test_ragged_array.py]: Same to me... I'll change it...

**Review by jorisvandenbossche — APPROVED** [review]
Thanks!

**Pull request merged**